versioned/mongodb/src/test/java/com/dremio/nessie/versioned/store/mongodb/SampleEntities.java

```diff
@@ -0,0 +1,208 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.mongodb;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.dremio.nessie.versioned.impl.Fragment;
+import com.dremio.nessie.versioned.impl.InternalCommitMetadata;
+import com.dremio.nessie.versioned.impl.InternalRef;
+import com.dremio.nessie.versioned.impl.InternalValue;
+import com.dremio.nessie.versioned.impl.L1;
+import com.dremio.nessie.versioned.impl.L2;
+import com.dremio.nessie.versioned.impl.L3;
+import com.dremio.nessie.versioned.store.Entity;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * This utility class generates sample objects mapping to each enumerate in
+ * {@link com.dremio.nessie.versioned.store.ValueType}.
+ * It is intended to be a central place for test data generation used by multiple test suites.
+ */
+public class SampleEntities {
+  private static final Entity ID = Entity.ofBinary(
+      new byte[] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19});
+  private static final Entity ID2 = Entity.ofBinary(
+      new byte[] {19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0});
+
+  /**
+   * Create a Sample L1 entity.
+   * @return sample L1 entity.
+   */
+  public static L1 createL1() {
+    final Map<String, Entity> keyList = new HashMap<>();
+    keyList.put("chk", Entity.ofBoolean(true));
+    keyList.put("fragments", Entity.ofList(ID, ID2, ID));
+    keyList.put("mutations", Entity.ofList(Entity.ofMap(ImmutableMap.of("a",
+        Entity.ofList(Entity.ofString("addition1"), Entity.ofString("addition2"))))));
+
+    final List<Entity> treeList = new ArrayList<>(L1.SIZE);
+    for (int i = 0; i < L1.SIZE; i++) {
+      treeList.add(ID);
+    }
+
+    final Map<String, Entity> attributeMap = new HashMap<>();
+    attributeMap.put("metadata", ID);
+    attributeMap.put("tree", Entity.ofList(treeList));
+    // The ID is generated from the contents of the L1 object. If any of the contents change, the ID will as well.
+    attributeMap.put("id", Entity.ofBinary(
+        new byte[] {110, 40, 7, 86, -103, 23, 6, -34, -102, 95, -33, 77, -87, 31, 23, 110, 67, -46, 56, -45}));
+    attributeMap.put("keys", Entity.ofMap(keyList));
+    attributeMap.put("parents", Entity.ofList(ID2));
+
+    return L1.SCHEMA.mapToItem(attributeMap);
+  }
+
+  /**
+   * Create a Sample L2 entity.
+   * @return sample L2 entity.
+   */
+  public static L2 createL2() {
+    final List<Entity> treeList = new ArrayList<>(L2.SIZE);
+    for (int i = 0; i < L2.SIZE; i++) {
+      treeList.add(ID);
```

Comment:
using the same id for every position doesn't do a great job of ensuring validation on encoding, etc.